pandas/core/indexes/multi.py

```diff
@@ -2862,9 +2862,17 @@ def _maybe_to_slice(loc):
         # needs linear search within the slice
         i = self._lexsort_depth
         lead_key, follow_key = key[:i], key[i:]
-        start, stop = (
-            self.slice_locs(lead_key, lead_key) if lead_key else (0, len(self))
-        )
+
```

Comment:
hmm why not handle in .slice_locs itself?

Reply:
not sure yet

Reply:
kk lmk on this, i think it would be nicer to the caller

Reply:
id like to address in a follow-up, avoid merge conflict with other MultiIndex branches